src/sync/pull-request.ts

```diff
@@ -57,18 +61,25 @@ export default async function(
 	const asyncTags:any[] = [];
 	// TODO: use graphql here instead of rest API
 	await statsd.asyncTimer(
+		// Retry up to 6 times pausing for 10s, for *very* large repos we need to wait a while for the result to succeed in dotcom
 		async () => {
-			({
-				data: edges,
-				status,
-				headers
-			} = await github.pulls.list({
-				...vars,
-				state: "all",
-				sort: "created",
-				direction: "desc"
-				// Retry up to 6 times pausing for 10s, for *very* large repos we need to wait a while for the result to succeed in dotcom
-			}));
+			(
+
+				{
+					data: edges,
+					status,
+					headers
+				} = useNewGHClient?
+					await github.pulls.list({ ...vars, state: "all", sort: "created", direction: "desc"})
+					:
+					await newGithub
+						.getPullRequests(repository.owner.login, repository.name, perPage, Number(cursor),
+							{
```

Comment:
I have one idea, hence we still have to pass all old parameters here, should we keep old and new client APIs the same, so we can simply replace the GitHub Client object and keep the rest of the code in these handlers as is?

Reply:
It is the same API in this case. But if the old API is crap, I wouldn't make the new API equally crap and instead add some mapping in this code here.